### PR TITLE
feat: support ai backends

### PR DESCRIPTION
**Pull request by jueli12**
## What type of PR is this?

/kind feature

## What this PR does / why we need it:

support an AI backend service that other modules can use to access AI language models.

### CODE REVIEW

**Comment by jueli12** [comment]
I have read the CLA Document and I hereby sign the CLA

File: cmd/app/options/ai.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package options
```
Comment (elliotxx) [inline cmd/app/options/ai.go:1]: You can execute `make fix-license` to batch supplement license comment in the project root directory.

File: go.mod [hunk 2]
```diff
@@ -64,6 +64,11 @@ require (
 	sigs.k8s.io/yaml v1.3.0
 )
 
+require (
+	github.com/hupe1980/go-huggingface v0.0.15 // indirect
+	github.com/sashabaranov/go-openai v1.27.0 // indirect
+)
```
Comment (elliotxx) [inline go.mod:70]: You can merge all `require` and then execute `go mod tidy` formatting, which normally only consists of two parts: direct and indirect.

**Review by elliotxx — APPROVED** [review]
lgtm

File: go.mod [hunk 3]
```diff
@@ -127,6 +128,7 @@ require (
 	github.com/prometheus/procfs v0.8.0 // indirect
 	github.com/rogpeppe/go-internal v1.6.1 // indirect
 	github.com/russross/blackfriday/v2 v2.1.0 // indirect
+	github.com/sashabaranov/go-openai v1.27.0
```
Comment (elliotxx) [inline go.mod:131]: These two packages should appear in the 'require' section above.

**Pull request merged**